03-iam/3.3/iam.yaml

```diff
@@ -0,0 +1,65 @@
+Description: 'IAM Role for stelligent-u'
+Metadata:
+  SupportedResources: 'IAM S3'
+Resources:
+  S3FullAccess:
+    Type: 'AWS::IAM::Role'
+    Properties:
+      Description: 'IAM role for Full S3 Access'
+      AssumeRolePolicyDocument:
+        Version: '2012-10-17'
+        Statement:
+          - Effect: Allow
+            Principal:
+              AWS: 'arn:aws:iam::324320755747:user/michael.mcelreath.labs'
+            Action:
+              - 'sts:AssumeRole'
+  RolePolicies:
+    Type: 'AWS::IAM::ManagedPolicy'
+    Properties:
+      ManagedPolicyName: 'mmcelreath-training-admin'
+      Description: 'IAM policy for full S3 access'
+      PolicyDocument:
+        Version: '2012-10-17'
+        Statement:
+          - Effect: Allow
+            Action: 
+              - "s3:Put*"
+            Resource: 
+              - !Join ['',['arn:aws:s3:::', !Ref s3Bucket, '/lebowski/*']]
+          - Effect: Allow
+            Action: 
+              - "s3:Get*"
+              - "s3:List*"
+            Resource: 
+              - !Join ['',['arn:aws:s3:::', !Ref s3Bucket, '/*']]
+              - !Join ['',['arn:aws:s3:::', !Ref s3Bucket]]
+          - Effect: Allow
+            Action: 
+              - "s3:Get*"
+              - "s3:List*"
+            Resource: 
+              - !Join ['',['arn:aws:s3:::', !Ref s3Bucket2, '/*']]
```

Comment:
In general the best practice for cfn names are CamelCase - I highly recommend you stick with that.